### PR TITLE
STAM-147 single sign out function

### PR DESCRIPTION
**Pull request by crazyrohila**
* Agent logout function
 - Session logout via API call
 - Clear in-memory cache
 - trigger openam/UI/Logout

### CODE REVIEW

File: src/amclient/am-client.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ export class AmClient {
     return Axios
       .get(`${this.serverAddress}/json/realms/root/agents/${agentId}`, {
         headers: {
-          host: this.hostname,
-          cookie: `${cookieName}=${sessionId}`
+          Host: this.hostname,
+          Cookie: `${cookieName}=${sessionId}`
```
Comment (niroz89) [inline src/amclient/am-client.ts:52]: HTTP headers are case in-sensitive.

**Review by niroz89 — COMMENTED** [review]
I'm not sure sending request to '/UI/Logout' will be the responsibility of node-openam-agent, ideally it should be performing only '/json/sessions?_action=logout' and deletes AM session. We have to send to /UI/Logout in order to perform a logout chain based on internal requirement. Let's discuss on this.

File: src/amclient/am-client.ts (hunk 1 above)
Comment (crazyrohila) [reply src/amclient/am-client.ts:52]: yeah, I just want to make it consistent with other headers' names which were having first letter capital.

**Comment by crazyrohila** [comment]
> I'm not sure sending request to '/UI/Logout' will be the responsibility of node-openam-agent, ideally it should be performing only '/json/sessions?_action=logout' and deletes AM session. We have to send to /UI/Logout in order to perform a logout chain based on internal requirement. Let's discuss on this.

I have tried that as well and have a function that does that, but in that case, we can clear cookie only for the current domain. It was able to delete the session (I have checked AM sessions).

**Comment by crazyrohila** [comment]
@niroz89 I have updated the logout function to clear the session via and removed the AM logout from SDK, the application will build that itself.

**Comment by riteshgurung** [comment]
Signout working. Need a separate PR for Policy shield

**Pull request merged**